package NYT.NTabletClient.NProto;

import "yt/core/misc/guid.proto";
import "yt/ytlib/hive/transaction_supervisor_service.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqStartTransactionExt
{
    extend NYT.NHive.NProto.TReqStartTransaction
    {
        optional TReqStartTransactionExt start_transaction_ext = 101;
    }

    // Transaction timeout (in ms).
    optional int32 timeout = 1;

    // NB: Tablet transaction ids are generated by clients.
    optional NYT.NProto.TGuid transaction_id = 2;
}

////////////////////////////////////////////////////////////////////////////////

